Libraries/LibWeb/HTML/Storage.cpp

```diff
@@ -35,9 +45,14 @@ Storage::Storage(JS::Realm& realm, Type type, u64 quota_bytes)
         .named_property_setter_has_identifier = true,
         .named_property_deleter_has_identifier = true,
     };
+
+    all_storages().set(*this);
 }
 
-Storage::~Storage() = default;
+Storage::~Storage()
+{
+    all_storages().remove(*this);
```

Comment:
This should be in `Storage::finalize()` to ensure that other cells being destroyed by the same GC pass won't find their way to this `Storage` irrespective of destruction order.

Reply:
Ah, thanks! Sounds like I should apply the same fix here to the same pattern in `MessagePort`!

Reply:
Oh yeah definitely :)

Reply:
I was thinking about this a little bit more as I implement this - is there a case to be made that we should consider disallowing destructors for GC::Cell based objects in the clang plugin?

Reply:
Hmm, maybe(?)

We still need destructors, otherwise GC::Cell subclass couldn't have members with destructors, e.g AK::String etc.

But perhaps there's a case for always preferring finalize() over an explicitly defined destructor?

Reply:
Yeah, that's the train of thought I was going down - technically I guess it couldn't solve all potential problems that finalize is trying to solve since a GC::Cell may have a non-Cell based member variable which has a destructor that pokes into a reference to another GC::Cell that may or may not be destroyed at that time.

But I suppose forcing all Cell based types not have a user defined destructor and force finalize should automatically catch what seems like would be the more common case of someone like me who wasn't aware of that potential footgun of GC types. 🤔

Reply:
Yeah the more I think about it, the more reasonable it seems 🤔